clar2wasm/src/wasm_generator.rs

```diff
@@ -3729,962 +3031,153 @@ impl WasmGenerator {
-    }
-
-    fn visit_static_contract_call(
-        &mut self,
-        _builder: &mut InstrSeqBuilder,
         _expr: &SymbolicExpression,
-        _contract_identifier: &QualifiedContractIdentifier,
-        _function_name: &ClarityName,
-        _args: &[SymbolicExpression],
+        input: &SymbolicExpression,
+        throws: &SymbolicExpression,
     ) -> Result<(), GeneratorError> {
-        Ok(())
+        self.traverse_expr(builder, input)?;
+        self.traverse_expr(builder, throws)
     }
 
-    fn traverse_dynamic_contract_call(
+    fn traverse_unwrap_panic(
         &mut self,
         builder: &mut InstrSeqBuilder,
         expr: &SymbolicExpression,
-        trait_ref: &SymbolicExpression,
-        function_name: &ClarityName,
-        args: &[SymbolicExpression],
-    ) -> Result<(), GeneratorError> {
-        self.traverse_expr(builder, trait_ref)?;
-        for arg in args.iter() {
-            self.traverse_expr(builder, arg)?;
-        }
-        self.visit_dynamic_contract_call(builder, expr, trait_ref, function_name, args)
-    }
-
-    fn visit_dynamic_contract_call(
-        &mut self,
-        _builder: &mut InstrSeqBuilder,
-        _expr: &SymbolicExpression,
-        _trait_ref: &SymbolicExpression,
-        _function_name: &ClarityName,
-        _args: &[SymbolicExpression],
+        input: &SymbolicExpression,
     ) -> Result<(), GeneratorError> {
-        Ok(())
+        self.traverse_expr(builder, input)?;
+        self.visit_unwrap_panic(builder, expr, input)
     }
 
-    fn traverse_contract_of(
+    fn traverse_stx_burn(
         &mut self,
         builder: &mut InstrSeqBuilder,
         expr: &SymbolicExpression,
-        name: &SymbolicExpression,
+        amount: &SymbolicExpression,
+        sender: &SymbolicExpression,
     ) -> Result<(), GeneratorError> {
-        self.traverse_expr(builder, name)?;
-        self.visit_contract_of(builder, expr, name)
+        self.traverse_expr(builder, amount)?;
+        self.traverse_expr(builder, sender)?;
+        self.visit_stx_burn(builder, expr, amount, sender)
     }
 
-    fn visit_contract_of(
+    fn traverse_stx_transfer(
         &mut self,
-        _builder: &mut InstrSeqBuilder,
-        _expr: &SymbolicExpression,
-        _name: &SymbolicExpression,
+        builder: &mut InstrSeqBuilder,
+        expr: &SymbolicExpression,
+        amount: &SymbolicExpression,
+        sender: &SymbolicExpression,
+        recipient: &SymbolicExpression,
+        memo: Option<&SymbolicExpression>,
     ) -> Result<(), GeneratorError> {
-        Ok(())
+        self.traverse_expr(builder, amount)?;
+        self.traverse_expr(builder, sender)?;
+        self.traverse_expr(builder, recipient)?;
+        if let Some(memo) = memo {
+            self.traverse_expr(builder, memo)?;
+        }
+        self.visit_stx_transfer(builder, expr, amount, sender, recipient, memo)
```

Comment:
In this new model, should we merge the `visit_` methods into the `traverse` methods?

Reply:
Let's postpone that decision